### PR TITLE
Masterclasses: use the scarcity effect

### PR DESCRIPTION
**Pull request by regiskuckaertz**
## What does this change?

When fewer than 10% of tickets are left for a masterclass, the component will warn the user the event is about to sell out:
picture 1
## What is the value of this and can you measure success?

A test conducted by the merch team has shown increased conversion when using this psychological trigger. This is no different than your average ecommerce site saying there's only 3 items left so you'd better hurry.
## Does this affect other platforms - Amp, Apps, etc?

Nope.
## Request for comment

@kelvin-chappell can you please review the awesome Scala code I wrote?

### CODE REVIEW

File: static/src/stylesheets/module/commercial/_lineitem.scss [hunk 1]
```diff
@@ -72,6 +72,9 @@
     .lineitem__cta {
         margin-top: $gs-baseline/2;
     }
+    .lineitem__scarcity {
+        color: guss-colour(rainbow-red);
```
Comment (uplne) [inline static/src/stylesheets/module/commercial/_lineitem.scss]: unicorn

File: commercial/app/model/commercial/masterclasses/MasterClass.scala [hunk 2]
```diff
@@ -99,12 +101,14 @@ case class EventbriteMasterClass(id: String,
     } else f"£${priceList.head}%,.2f"
   }
 
+  lazy val ratioTicketsLeft = 1 - (tickets.map(_.quantitySold).reduceLeft(_ + _) / tickets.map(_.quantityTotal).reduceLeft(_ + _))
```
Comment (kelvin-chappell) [inline commercial/app/model/commercial/masterclasses/MasterClass.scala]: `.reduceLeft(_ + _)` could just be `.sum`

Comment (kelvin-chappell) [inline commercial/app/model/commercial/masterclasses/MasterClass.scala]: Perhaps you could add a test for this, to avoid regression?

**Comment by kelvin-chappell** [comment]
Awesome scala code is looking good! :trophy: 

When comments considered, :+1:

File: commercial/app/model/commercial/masterclasses/MasterClass.scala (hunk 2 above)
Comment (regiskuckaertz) [inline commercial/app/model/commercial/masterclasses/MasterClass.scala]: Yes, good idea!

**Pull request merged**